src/core/namespace.c

```diff
@@ -2540,6 +2637,20 @@ int setup_namespace(const NamespaceParameters *p, char **reterr_path) {
                         return r;
         }
 
+        if (unshare(CLONE_NEWNS) < 0) {
```

Comment:
I do not think this should be moved.
Rather than move this, how about to call bpffs_finalize() and friends in mount_bpffs() ?

Reply:
done, in this way I don't leak the fs anymore